### PR TITLE
[TORCH_TRACE] Record stack when no compile context is available

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122644

This will help me track down those annoying unknown compile products.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: torch/_logging/_internal.py [hunk 1]
```diff
@@ -1063,6 +1066,10 @@ def trace_structured(
                 record["frame_id"] = trace_id.compile_id.frame_id
                 record["frame_compile_id"] = trace_id.compile_id.frame_compile_id
                 record["attempt"] = trace_id.attempt
+            else:
+                # Record the stack of the log call to better diagnose why we
+                # don't have a frame id for it
+                record["stack"] = torch._logging.structured.from_traceback(CapturedTraceback.extract(skip=1).summary())
```
Comment (jamesjwu) [inline torch/_logging/_internal.py:1072]: OOC, how do we reconstruct the stack for the trie when the compile id is known? Is there some mapping we get from the frame_id?

**Review by jamesjwu — APPROVED** [review]
LGTM

File: torch/_logging/_internal.py (hunk 1 above)
Comment (ezyang) [reply torch/_logging/_internal.py:1072]: When we start compilation, we dump the user stack trace of the code that triggered the compile. Those stacks get put together.

**Pull request closed without merging**